### PR TITLE
[CDE-1042] - Popup on the marker in NewMapComponent not working with engine as "google"

### PR DESCRIPTION
**Pull request by bcostahitachivantara**
@pentaho-lmartins 
@ssamora 
@ppatricio 
@smmribeiro

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **392**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/cde-plugin/job/PR-975/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: components/impls/core/src/main/javascript/components/NewMapComponent/amd/engines/google/MapEngineGoogle.js [hunk 1]
```diff
@@ -716,59 +716,70 @@ define([
...
+      _.each(this._popups, function (p) {
         p.setMap(null);
       });
       this._popups.push(popup);
     },
 
-    showPopup: function(data, feature, popupHeight, popupWidth, contents, popupContentDiv, borderColor) {
+    showPopup: function (data, feature, popupHeight, popupWidth, contents, popupContentDiv, borderColor) {
+      // Remove old popups div
+      var oldPopup = document.getElementById(popupContentDiv);
```
Comment (dcleao) [inline components/impls/core/src/main/javascript/components/NewMapComponent/amd/engines/google/MapEngineGoogle.js:752]: Possibly better named `oldContents`?

File: components/impls/core/src/main/javascript/components/NewMapComponent/amd/engines/google/MapEngineGoogle.js [hunk 2]
```diff
@@ -716,59 +716,70 @@ define([
...
+      if (oldPopup) {
+        oldPopup.remove();
+      }
+
+      if (!contents) {
+        // Create a new div for new popup info window
+        contents = document.createElement("div");
+        contents.setAttribute("id", popupContentDiv);
+      }
```
Comment (dcleao) [inline components/impls/core/src/main/javascript/components/NewMapComponent/amd/engines/google/MapEngineGoogle.js:761]: Should leave an empty line after statement blocks, for readability.

File: components/impls/core/src/main/javascript/components/NewMapComponent/amd/engines/google/MapEngineGoogle.js [hunk 3]
```diff
@@ -716,59 +716,70 @@ define([
...
       this._popups = this._popups || [];
-      _.each(this._popups, function(p) {
+      _.each(this._popups, function (p) {
         p.setMap(null);
       });
       this._popups.push(popup);
     },
 
-    showPopup: function(data, feature, popupHeight, popupWidth, contents, popupContentDiv, borderColor) {
+    showPopup: function (data, feature, popupHeight, popupWidth, contents, popupContentDiv, borderColor) {
```
Comment (dcleao) [inline components/impls/core/src/main/javascript/components/NewMapComponent/amd/engines/google/MapEngineGoogle.js:750]: Ideally, you'd rename the `popupContentDiv` parameter to `popupContentId`.
Additionally, `contents` could be renamed to `popupContent`.
The naming, as it is, is quite confusing.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **392**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/cde-plugin/job/PR-975/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**